ludwig/datasets/ohsumed/ohsumed.py

```diff
@@ -0,0 +1,74 @@
+import csv
+import pandas as pd
+from typing import Dict
+from ludwig.datasets.base_dataset import BaseDataset
+import shutil
+import os
+
+
+"""A class to process the ohsumed training data"""
+import yaml
+
+
+class OhsuMed(BaseDataset):
+
+    def __init__(self):
+        self.__initial_path = os.path.abspath(os.path.dirname(__file__))
+        self.__source_location =  os.path.join(self.__initial_path , "../text/ohsumed/ohsumed-allcats.csv")
+        self.__cache_location = os.path.join(self.__initial_path, "../../../.ludwig_cache/")
+        self.__config_file_path = os.path.join(self.__initial_path , "../text/versions.yaml")
+        with open(self.__config_file_path) as config_file:
+            self.__config_file_contents = yaml.load(config_file, Loader=yaml.FullLoader)
+        cur_version = self.__config_file_contents["text"]["ohsumed"]
+        self.__dest_location = self.__cache_location + "ohsumed_" + str(cur_version) + ".csv"
+        self.__result_dict={}
+
+    """Download the raw data to the ludwig cache in the format ~/.ludwig_cache/id
+    where is is represented by the name.version of the dataset
+    args:
+        self (BaseDataset): A pointer to the current class
+        dataset_name: The name of the dataset we need to retrieve
+    return:
+        true or false depending on whether the file exists in the new location
+    """
+    def download(self, dataset_name) -> bool:
```

Comment:
The parent interface suggests this should not return anything.  I think having no return value makes more sense here.  If the dataset cannot be downloaded, we should raise an exception.  What do you think?

Reply:
Changed all methods other than load to return nothing